sp/src/game/server/hl2/npc_zombine.cpp

```diff
@@ -1802,6 +1845,380 @@ void CNPC_MetroZombie::ReadyGrenadeSound( void )
+// Purpose: Play a sound while charging towards the enemy
+//-----------------------------------------------------------------------------
+void CNPC_HEVZombie::ChargeSound( void )
+{
+	switch (m_tEzVariant)
+	{
+	case EZ_VARIANT_RAD:
+		EmitSound( "HEVGlowbie.Charge" );
+		EmitSound( "HEVGlowbie.Vox_Charge" ); // This soundscript should be on a different channel
+		break;
+	case EZ_VARIANT_XEN:
+		EmitSound( "HEVXenbie.Charge" );
+		EmitSound( "HEVXenbie.Vox_Charge" ); // This soundscript should be on a different channel
+		break;
+	default:
+		EmitSound( "HEVZombie.Charge" );
+		EmitSound( "HEVZombie.Vox_Charge" ); // This soundscript should be on a different channel
+		break;
+	}
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Play a sound while readying a grenade
+//-----------------------------------------------------------------------------
+void CNPC_HEVZombie::ReadyGrenadeSound( void )
+{
+	switch (m_tEzVariant)
+	{
+	case EZ_VARIANT_RAD:
+		EmitSound( "HEVGlowbie.ReadyGrenade" );
+		break;
+	case EZ_VARIANT_XEN:
+		EmitSound( "HEVXenbie.ReadyGrenade" );
+		break;
+	default:
+		EmitSound( "HEVZombie.ReadyGrenade" );
+		break;
+	}
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: HEV Zombies have suit charge
+//-----------------------------------------------------------------------------
+int CNPC_HEVZombie::OnTakeDamage( const CTakeDamageInfo &inputInfo )
+{
+	CTakeDamageInfo info = inputInfo;
+
+	// Armor code taken from CNPC_CloneCop::OnTakeDamage
+	if (info.GetDamage() && m_ArmorValue && !(info.GetDamageType() & (DMG_FALL | DMG_DROWN | DMG_POISON | DMG_RADIATION)))// armor doesn't protect against fall or drown damage!
+	{
+		float flRatio = 0.2f;
+
+		float flNew = info.GetDamage() * flRatio;
+
+		float flArmor;
+
+		flArmor = (info.GetDamage() - flNew);
+
+		if (flArmor < 1.0)
+		{
+			flArmor = 1.0;
+		}
+
+		// Does this use more armor than we have?
+		if (flArmor > m_ArmorValue)
+		{
+			flArmor = m_ArmorValue;
+			flNew = info.GetDamage() - flArmor;
+			m_ArmorValue = 0;
+		}
+		else
+		{
+			m_ArmorValue -= flArmor;
+		}
+
+		info.SetDamage( flNew );
+	}
+
+	return BaseClass::OnTakeDamage( info );
+}
+
+extern ConVar sk_battery; // Need to know how much battery is applied per item
+
+//-----------------------------------------------------------------------------
+// Purpose: HEV Zombies should drop a bunch of items on death
+// TODO - SHould we be worried about the variant of the item?
+//-----------------------------------------------------------------------------
+void CNPC_HEVZombie::Event_Killed( const CTakeDamageInfo &info )
+{
+	BaseClass::Event_Killed( info );
+
+	if (m_bDropItems) {
+		int numBatteries = MAX( m_ArmorValue / sk_battery.GetInt(), random->RandomInt( 1, 3 ));
+		for (int i = 0; i < numBatteries; i++)
+		{
+			DropItem( "item_battery", WorldSpaceCenter() + RandomVector( -4, 4 ), RandomAngle( 0, 360 ) );
```

Comment:
This should match the HEV zombie's E:Z variant

Reply:
Doesn't DropItem() do that already? I thought it was overridden in basenpc

Reply:
Whoops! You're right, sorry 😅